src/hotspot/share/opto/addnode.hpp

```diff
@@ -78,6 +78,13 @@ class AddNode : public Node {
   virtual int min_opcode() const = 0;
 
   static AddNode* make(Node* in1, Node* in2, BasicType bt);
+
+  // Utility function to check if the given node is a NOT operation,
+  // i.e., n == m ^ (-1).
+  static bool is_not(PhaseGVN* phase, Node* n, BasicType bt);
```

Comment:
Could these be made non-static?

Reply:
Hmm, I agree with this idea. `n->is_not(...)` would really be nicer. You'd probably have to move the two methods to `node.hpp`.

Reply:
@TobiHartmann @eme64 
I moved `is_not` but I was not able to move `make_not` to `node` class, because otherwise it would not compile for arm, s390x, ppc64le.
```
/home/runner/work/jdk/jdk/src/hotspot/share/opto/node.cpp:1605:18: error: expected type-specifier before 'XorINode'
 1605 |       return new XorINode(this, phase->intcon(-1));
```
Please let me know if we still want to move `make_not`. Thanks.

Reply:
I would say it's better to leave both methods as static methods then, for consistency. Thanks for giving it a try!